src/external/dftk_to_wannier90.jl

```diff
@@ -0,0 +1,605 @@
+using ProgressMeter
+using LinearAlgebra
+using Dates
+
+## Works in the following order
+# $ dftk2wan_win_file(...) to generate the win file
+# $ wannier90.x -pp "prefix" ------------ nnkp file
+# $ dftk2wan_wannierization_files(...)--- mmn,amn,eig files
+# $ wannier90.x "prefix"  --------------- actual wannierization
+
+## Remark : the routines "dict_AOs", "guess_win", "cart_to_spherical" and "A_k_matrix_win_guesses" and the "projs" table in "read_nnkp_file"
+# are dedicated to computation of A_k matrices with Wannier90 guess, read from the nnkp file.
+# It is bugged and may not be usefull but I left it there.
+
+## TODO
+# 1 - check if kgrid can be removed from dftk2wan_win_file arguments
+# 2 - add handling of the SCDM routine as an alternative guess
+# 3 - optimize the "generate_mmn" routine
+
+
+"""
+    Create a .win file for Wannier90, compatible with the system studied with DFTK.
+    Options have to be compeleted by hand.
+"""
+function dftk2wan_win_file(prefix::String,basis::PlaneWaveBasis,scfres,num_wann;
+                           kgrid = [4,4,4], #can we retrieve this from scfres or basis ? Feel this should not be an argument.
+                           bands_plot = false,
+                           num_iter = 500,
+                           num_print_cycles = 50
+                           )
+    
+    ## Check kgrids
+    if (prod(kgrid) != size(scfres.ψ,1))
+        error("The given kgrid doesn't match the one used for the scf calculation")
+    end    
+
+    ## Write in file   
+    open("$prefix.win","w") do f
+
+        write(f,"! $prefix.win file generated by DFTK at "*string(now())*"\n"^2)
+        
+        ## General parameters
+        num_bands = size(scfres.ψ[1],2)
+        write(f,"num_bands "*(" "^9)*"= $num_bands"*"\n")
+        write(f,"num_wann "*(" "^10)*"= $num_wann"*"\n"^2)
+
+        write(f,"num_iter "*(" "^10)*"= $num_iter"*"\n")
+        write(f,"num_print_cycles "*(" "^2)*"= $num_print_cycles"*"\n"^2)
+        
+        ## Messages for the user
+        write(f,"!"^20*" Complete with optional parameters :"*"\n"^2)
+
+
+
+        ## System
+        write(f,"!"^20*" System"*"\n"^2)
+
+        # Unit cell block
+        write(f,"begin unit_cell_cart"*"\n"*"bohr"*"\n")
+        unit_cell = transpose(basis.model.lattice)         #unit cell vectors are in raws in Wannier90.
```

Comment:
are units OK here?

Reply:
They are here because I entered the lattice vectors in cartesian coordinates and in bohr. 
But that's a good point I thought this was the standard way to do in DFTK but if there are other possible units and coordinates we should take this into account. For now it only works with Bohr + cartesian.

Reply:
No in DFTK lattice vectors are always Bohr (and in columns).